src/urllib3/util/retry.py

```diff
@@ -227,6 +296,23 @@ def new(self, **kw):
             remove_headers_on_redirect=self.remove_headers_on_redirect,
             respect_retry_after_header=self.respect_retry_after_header,
         )
+
+        # TODO: If already given in **kw we use what's given to us
+        # If not given we need to figure out what to pass. We decide
+        # based on whether our class has the 'method_whitelist' property
+        # and if so we pass the deprecated 'method_whitelist' otherwise
+        # we use 'method_allowlist'. Remove in v2.0
+        if "method_whitelist" not in kw and "method_allowlist" not in kw:
+            if "method_whitelist" in self.__dict__:
```

Comment:
I'm doing this instead of `hasattr(self, "method_whitelist")` because that triggers a `getattr()` call. [Thanks Hynek for the blog post](https://hynek.me/articles/hasattr/).